chapter1/rotateMatrix.js

```diff
@@ -0,0 +1,44 @@
+// Given an image represented by an NxN matrix, where each pixel in the image is 4 bytes, write a method to rotate the image by 90 degrees.
+
+/*
+Input:
+[
+  [1,2,3],
+  [4,5,6],
+  [7,8,9]
+];
+Output: 
+[
+  [7,4,1],
+  [8,5,2],
+  [9,6,3]
+];
+
+*/
+rotateMatrix = (arr) => {
+    var rotateArr = [];
+    var arrLen = arr.length;
+    var  i = 0;
+
+    function loopThrough(i){
+        if (rotateArr.length === arrLen ){
+            return rotateArr;
+        }  
+
+        while(i < arr.length){
```

Comment:
Using `arr.length` instead of `arrLen` variable declared above will cause the `.length` property lookup to happen every iteration which can hurt performance for very large inputs (these "micro-optimizations" add up over time for real savings).